### DISCUSSION TITLE
Window right snap occasionally stops working.

### DISCUSSION THREAD

**Discussion by bdmorin** [rxhanson/RectanglePro-Community#340] [General]
I cannot figure out what condition it is that causes this, however sometimes my windows don't fully snap to the right. 

![CleanShot 2023-08-23 at 12 21 57](https://github.com/rxhanson/RectanglePro-Community/assets/476219/3a6965be-e818-4a57-b5c1-343dc4690daa)
![CleanShot 2023-08-23 at 12 23 58](https://github.com/rxhanson/RectanglePro-Community/assets/476219/87343b19-64dd-41ce-b375-41afe95d12a2)

- I have to quit Rectangle Pro, and start it again, and sometimes that doesn't help. 
- I may have an unusual setup.
- My macbook is connected to a pluggable dock that gives me 3 monitors. 
  - Macbook -- Monitor1 -- Monitor 2 -- Monitor 3
  - So I have 4 screens total, and I use the [displaylink](https://support.displaylink.com/knowledgebase/topics/80209-troubleshooting-macos) driver to make it all work.
![CleanShot 2023-08-23 at 12 25 51](https://github.com/rxhanson/RectanglePro-Community/assets/476219/d4e1cafd-3529-4f11-b682-ea1e2ee00df9)

Sometimes I'll close my laptop lid and it works again without restarting RP; other times no combination works. 

Anyone else experiencing this?

**Comment by rxhanson** [comment]
Thanks for reporting. Rectangle Pro relies on macOS to tell it the usable dimensions of each screen. Since this appears to be sporadic, it's likely here that this issue is out of Rectangle Pro's control to obtain the proper reading on the dimensions (my guess is that this would be due to the display link driver).

Otherwise, one thing to check is to make sure that pin mode is disabled. 

To dig deeper, we'll need logs: hold the option key in the menu bar menu and select "View Logging..."

Probably not necessary, but I can load your config and see if anything jumps out at me. You can export it in the settings tab of the prefs window.

**Comment by bdmorin** [comment]
Sounds good, I'll post later. Thanks!

**Comment by bdmorin** [comment]
So - I've been able to find a common factor when this situation happens, it's some kind of positioning issue with 1password. I'm going to block rectangle from 1password to see if it helps.
